### ISSUE TITLE
Potential speed up for the data fetchers

### ISSUE THREAD

**Issue by arokem** [dipy/dipy#2494]
We noticed that the data fetchers take a long time to run, even if the data is already available on the user's hard-drive. I think that the reason for that is that we validate the file content's md5 every time we call the fetcher: https://github.com/dipy/dipy/blob/be956a529465b28085f8fc435a756947ddee1c89/dipy/data/fetcher.py#L143. It might save a little bit of time to not do this validation every time, but then if the file was corrupted since it was downloaded, the user will not know until some downstream consequence hits them. Does anyone have any thoughts about this trade-off?

**Comment by skoudoro** [comment]
Hi @arokem,

Thanks for the feedback. Can you provide some timing? 

Every day, we have a lot of new users discovering DIPY, so I would recommend keeping this check even if it slows down the process.
However, we can provide a flag for the advanced users to skip this md5 check. Easy feature to add and we keep the current behavior. What do you think?

**Comment by arokem** [comment]
Hmm. Actually, this doesn't seem too terrible: 

```
In [3]: %timeit _get_file_md5("/Users/arokem/.dipy/stanford_hardi/HARDI150.nii.g
   ...: z")
173 ms ± 11.6 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

```
In [5]: %timeit dpd.fetch_stanford_hardi()
173 ms ± 10.4 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

So, this is a <200 msec overhead, which almost entirely spent on getting that hash. 

And also: 

```
In [6]: %timeit dpd.read_stanford_hardi()
181 ms ± 10.4 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Which means that when you read the files (lazy loading on the nii.gz!), you are still not incurring much more overhead than that. I would still think that adding an option not to hash the file contents could be good to have, so I'd be +1 for adding that as an option, defaulting to the "beginner mode".

@36000 : given these timings, is this really the delay you have observed?

**Comment by skoudoro** [comment]
Hi @arokem and @36000,

is it ok to close this issue?

Timings are ok, and I do not see so much benefit to add an option to skip the md5. 

Please let me know

**Comment by skoudoro** [comment]
finally, I think it is better if I close it to remove you this burden.

feel free to reopen this issue in case you want the "skip the md5" implemented.

**Comment by 36000** [comment]
Yes, we should have closed this issue a while ago, the timing was something else. Thank you!
